<div style="
    background-color:#3fc1c9;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">NIVEL 1
    
</div>

<div style="
    background-color:#cfe6ff;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">
  <span style="display:block;">1. Consulta la API pública JSONPlaceholder utilizando el método GET para obtener:</span>
  <span style="display:block;">• Lista de publicaciones (/posts)</span>
  <span style="display:block;">• Lista de usuarios (/users)</span>
  <span style="display:block;">• Lista de tareas (/todos)</span>
</div>

In [ ]:
# Imports y configuración

import requests
import pandas as pd

BASE_URL = "https://jsonplaceholder.typicode.com"

#Para que pandas muestre mejor los DataFrames

pd.set_option("display.max_columns", 50)
pd.set_option("display.width", 120)

In [3]:
# GET posts
response_posts = requests.get(BASE_URL + "/posts")
posts = response_posts.json()

# GET users
response_users = requests.get(BASE_URL + "/users")
users = response_users.json()

# GET todos
response_todos = requests.get(BASE_URL + "/todos")
todos = response_todos.json()

In [7]:
df_posts = pd.DataFrame(posts)
df_users = pd.DataFrame(users)
df_todos = pd.DataFrame(todos)

<div style="
    background-color:#cfe6ff;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">
  <span style="display:block;">2. Muestra por pantalla:</span>
  <span style="display:block;">• La cantidad total de cada recurso.</span>
  <span style="display:block;">• El código de estado de cada petición.</span>
</div>

In [12]:
df_posts.shape

(100, 4)

In [13]:
df_users.shape

(10, 8)

In [14]:
df_todos.shape

(200, 4)

In [5]:
print("POSTS total:", len(posts))
print("Status posts:", response_posts.status_code)

print("USERS total:", len(users))
print("Status users:", response_users.status_code)

print("TODOS total:", len(todos))
print("Status todos:", response_todos.status_code)

POSTS total: 100
Status posts: 200
USERS total: 10
Status users: 200
TODOS total: 200
Status todos: 200


He hecho una petición GET a la API JSONPlaceholder para obtener posts, usuarios y tareas.
La respuesta se recibió en formato JSON y se convirtió en listas de Python. Tambien la he guardado en Data Frame para facilitar su analisis. Con funcion SHAPE podemos ver numero de registros. Despues he compraobado el status code de cada petición (200 = correcta) y he mostrado la cantidad total de registros de cada recurso (len).

<div style="
    background-color:#cfe6ff;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">
  <span style="display:block;">3. Realiza una petición a una publicación inexistente para obtener un error 404</span>
  <span style="display:block;">y muestra el código de estado recibido.</span>
</div>

In [ ]:
# Peticion 
response_404 = requests.get(BASE_URL + "/posts/947579")

print("Status code:", response_404.status_code)

Status code: 404


He realizado una petición GET a una publicación inexistente.
La API devolvió un código de estado 404 (Not Found), indicando que no existe.

<div style="
    background-color:#cfe6ff;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">
  <span style="display:block;">4. Realiza una petición POST para crear una nueva publicación ficticia.</span>
  <span style="display:block;">Incluye el título, el cuerpo del mensaje y un userId.</span>
  <span style="display:block;">• Muestra la respuesta JSON.</span>
  <span style="display:block;">• Muestra el código de estado.</span>
</div>

In [16]:
# Creo diccionario con el post:
new_post = {
    "title": "Peli mala",
    "body": "Esta pelicula esta fatal, lo mas aburrido que he visto en mi vida!",
    "userId": 1986
}

In [ ]:
# Envio a la API:
response_post = requests.post(BASE_URL + "/posts", json=new_post)

In [18]:
print(response_post.json())

{'title': 'Peli mala', 'body': 'Esta pelicula esta fatal, lo mas aburrido que he visto en mi vida!', 'userId': 1986, 'id': 101}


In [19]:
print("Status code:", response_post.status_code)

Status code: 201


He reralizado una petición POST a la API para crear una nueva publicación ficticia.
He enviado los datos en formato JSON y la API ha devuelto el objeto creado junto con un código de estado 201 (creado correctamente).

<div style="
    background-color:#cfe6ff;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">
  <span style="display:block;">5. Realiza una petición PATCH para modificar parcialmente una publicación existente.</span>
  <span style="display:block;">• Muestra la respuesta JSON.</span>
  <span style="display:block;">• Muestra el código de estado.</span>
</div>

In [ ]:
# Modifico el titulo del post (ej: id 1)
update_post = {
    "title": "Título nuevo ahora con PATCH"
}

response_patch = requests.patch(BASE_URL + "/posts/1", json=update_post)

# Respuesta JSON
print(response_patch.json())

# Status code
print("Status code:", response_patch.status_code)

{'userId': 1, 'id': 1, 'title': 'Título nuevo ahora con PATCH', 'body': 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto'}
Status code: 200


He realizado una petición PATCH para modificar parcialmente una publicación existente (su titulo).
La API devolvió el objeto actualizado y un código de estado 200 (correcto.).

<div style="
    background-color:#cfe6ff;
    color:#000000;
    padding:16px;
    margin-bottom:16px;
">
  <span style="display:block;">6. Haz una petición DELETE sobre una publicación.</span>
  <span style="display:block;">• Muestra la respuesta JSON.</span>
  <span style="display:block;">• Muestra el código de estado.</span>
</div>

In [21]:
# Borro una publicación (ej. id = 1)
response_delete = requests.delete(BASE_URL + "/posts/1")

# Respuesta JSON
print(response_delete.json())

# Status code
print("Status code:", response_delete.status_code)

{}
Status code: 200


He hecho una petición DELETE para eliminar un post.
La API ha devuelto una respuesta sin infformacion - vacia y un código de estado 200 (correcto).